packages/legacy/core/App/utils/PINCreationValidation.ts

```diff
@@ -36,16 +38,21 @@ export const PINCreationValidations = (PIN: string, PINRules: PINValidationRules
       errorName: PINError.OddOrEvenSequenceValidation,
     } as PINValidationsType)
   }
-  if (PINRules.no_repeated_numbers) {
-    let noRepeatedNumbers = new RegExp(/(\d)\1{1,}/)
-    if (typeof PINRules.no_repeated_numbers === 'number') {
-      noRepeatedNumbers = new RegExp(`(\\d)\\1{${PINRules.no_repeated_numbers - 1},}`)
-    }
+
+  if ('number' === typeof(PINRules.no_repeated_numbers) && PINRules.no_repeated_numbers > 2 ) {
```

Comment:
When would the `> 2` check be needed at the same time as the string interpolation below is? Wouldn't there only be one case (3) where that condition is satisfied?